src/Orleans.Pipeline.Client/OrleansPipe.cs

```diff
@@ -15,9 +15,9 @@ internal class OrleansPipe<TToServer, TFromServer>(
     private readonly Channel<TToServer> _writer = Channel.CreateUnbounded<TToServer>();
     private readonly Channel<TFromServer> _reader = Channel.CreateUnbounded<TFromServer>();
 
-    private IOrleansPipeGrain<TToServer, TFromServer>? _grain;
+    [AllowNull] private IOrleansPipeGrain<TToServer, TFromServer> _grain;
```

Comment:
What is the advantage of [AllowNull] instead of nullable fields? (i need to look up this attribute, but you may have deeper insight)

Reply:
@BernhardPollerspoeck no advantage other than not having to use `!` like in `_grain!.Write(writeData)` and similar places when you know that the `_grain` field is initialized before the usage on those methods.